src/components/RichTextEditor/serializer/markdown/deserializer/deserialize.ts

```diff
@@ -12,32 +12,23 @@ import {
     ListItemNode,
     ListNode,
     MarkdownAstNode,
+    OptionType,
     ParagraphNode,
-    PartialOptionType,
     TextNode,
     ThematicBreakNode,
-    defaultNodeTypes,
 } from '../types';
 
-export default function deserialize<T extends InputNodeTypes>(node: MarkdownAstNode, options?: PartialOptionType) {
-    const types = {
-        ...defaultNodeTypes,
-        ...options?.nodeTypes,
-        heading: {
-            ...defaultNodeTypes.heading,
-            ...options?.nodeTypes?.heading,
-        },
-    };
+export default function deserialize<T extends InputNodeTypes>(node: MarkdownAstNode, options: OptionType) {
+    const types = options?.nodeTypes as InputNodeTypes;
 
     const linkDestinationKey = options?.linkDestinationKey ?? 'link';
     const imageSourceKey = options?.imageSourceKey ?? 'link';
     const imageCaptionKey = options?.imageCaptionKey ?? 'caption';
 
     let children: Array<DeserializedNode<T>> = [{ text: '' }];
 
-    const nodeChildren = node.children;
-    if (nodeChildren && Array.isArray(nodeChildren) && nodeChildren.length > 0) {
-        children = nodeChildren.flatMap((c: MarkdownAstNode) =>
+    if (hasNodeChildren(node)) {
+        children = node.children!.flatMap((c: MarkdownAstNode) =>
```

Comment:
```suggestion
        children = node.children.flatMap((c: MarkdownAstNode) =>
```
So no `!` required :)

Reply:
Updated